A32NX/html_ui/Pages/A32NX_Core/A32NX_TransitionAltitude.js

```diff
@@ -0,0 +1,345 @@
+class A32NX_TransitionAltitude {
+    init() {
+        console.log('A32NX_TransitionAltitude init');
+        this.currentDeparture = "";
+        this.currentArrival = "";
+        this.offline = false;
+        this.checkStart = true;
+        this.checkStartSec = 0;
+        this.testAirport = "";
+        this.trySendRequest();
+    }
+
+    update(_deltaTime, _core) {
+        if (this.offline === true) {
+            this.offlineTransAlt();
+        }
+        this.tryCheckAPI(_deltaTime);
+    }
+
+    async trySendRequest() {
+        await NXApi.getAirport("KLAX")
+            .then((data) => {
+                this.testAirport = data.transAlt;
+            });
+    }
+
+    tryCheckAPI(_deltaTime) {
+        if (this.offline === false && this.checkStart === true) {
+            this.checkStartSec += _deltaTime;
+        }
+        if ((this.checkStartSec >= 60 * 1000) && (this.testAirport === "")) {
+            this.offline = true;
+            this.checkStart = false;
+        } else if ((this.checkStartSec >= 60 * 1000) && (this.testAirport !== "")) {
+            this.offline = false;
+            this.checkStart = false;
+            this.checkStartSec = 0;
+        }
+    }
+
+    offlineTransAlt() {
+        const Departure = NXDataStore.get("PLAN_ORIGIN", "");
+        const Arrival = NXDataStore.get("PLAN_DESTINATION", "");
+        if (Departure !== "" && Arrival !== "") {
+            if (this.currentDeparture !== Departure) {
+                this.departureLogic(Departure);
+                this.currentDeparture = Departure;
+            }
+            if (this.currentArrival !== Arrival) {
+                this.arrivalLogic(Arrival);
+                this.currentArrival = Arrival;
+            }
+        }
+    }
+
+    departureLogic(airport) {
+        const airportDepart_1 = airport.substr(0,1);
```

Comment:
**thought**: I believe this will contain the first letter of the departure airport.
**suggestion**: Consider renaming it to make that clear (also thinking about the origin/destination comment earlier).

**nitpick**: formatting of arguments should have a space after the `,`: `(0, 1)`

Reply:
Ok

Reply:
Not solved.

Reply:
wtf also this outdated. i merge 2 function to 1.